django/db/models/fields/related_descriptors.py

```diff
@@ -502,23 +502,26 @@ def __call__(self, **kwargs):
             return manager_class(self.instance)
         do_not_call_in_templates = True
 
+        def _filter_queryset(self, queryset):
+            db = self._db or router.db_for_read(self.model, instance=self.instance)
+            empty_strings_as_null = connections[db].features.interprets_empty_strings_as_nulls
+            queryset._add_hints(instance=self.instance)
+            if self._db:
```

Comment:
It's orthogonal to this PR but this `self._db` in place of just `db` looks wrong.